src/incremental_reader.rs

```diff
@@ -49,9 +53,10 @@ where
     pub fn new(reader: R) -> Self {
         Self {
             reader: Box::pin(reader),
+            writer: None,
             lines: String::with_capacity(VEC_BUFFER_CAPACITY * LINE_BUFFER_CAPACITY),
             line: String::with_capacity(LINE_BUFFER_CAPACITY),
-            writer: None,
+            non_utf8: Vec::with_capacity(SPLIT_UTF8_CODEPOINT_CAPACITY),
```

Comment:
If you're never going to exceed 4 bytes here, would it be more efficient to use `&[u8; 4]` here instead of `Vec<u8>`?

Reply:
Or more correct, because you couldn't accidentally allocate more space.